### ISSUE TITLE
deepstate-manticore fails on OneOf and Primes examples

### ISSUE THREAD

**Issue by agroce** [trailofbits/deepstate#66]
```
> deepstate-manticore build/examples/Primes

...

INFO:deepstate.mcore:Running 2 tests across 1 workers
2018-07-13 18:19:48,561: [8028] m.c.executor:ERROR: Exception: 'State' object has no attribute 'solver'
Traceback (most recent call last):
  File "/root/.local/lib/python2.7/site-packages/manticore-0.1.10-py2.7.egg/manticore/core/executor.py", line 461, in run
    if not current_state.execute():
  File "/root/.local/lib/python2.7/site-packages/manticore-0.1.10-py2.7.egg/manticore/core/state.py", line 127, in execute
    result = self._platform.execute()
  File "/root/.local/lib/python2.7/site-packages/manticore-0.1.10-py2.7.egg/manticore/platforms/linux.py", line 2216, in execute
    self.current.execute()
  File "/root/.local/lib/python2.7/site-packages/manticore-0.1.10-py2.7.egg/manticore/core/cpu/abstractcpu.py", line 822, in execute
    self._publish('will_execute_instruction', self.PC, insn)
  File "/root/.local/lib/python2.7/site-packages/manticore-0.1.10-py2.7.egg/manticore/utils/event.py", line 118, in _publish
    self._publish_impl(_name, *args, **kwargs)
  File "/root/.local/lib/python2.7/site-packages/manticore-0.1.10-py2.7.egg/manticore/utils/event.py", line 134, in _publish_impl
    sink._publish_impl(_name, *args, **kwargs)
  File "/root/.local/lib/python2.7/site-packages/manticore-0.1.10-py2.7.egg/manticore/utils/event.py", line 134, in _publish_impl
    sink._publish_impl(_name, *args, **kwargs)
  File "/root/.local/lib/python2.7/site-packages/manticore-0.1.10-py2.7.egg/manticore/utils/event.py", line 132, in _publish_impl
    sink._publish_impl(_name, self, *args, **kwargs)
  File "/root/.local/lib/python2.7/site-packages/manticore-0.1.10-py2.7.egg/manticore/utils/event.py", line 126, in _publish_impl
    callback(robj(), *args, **kwargs)
  File "/root/.local/lib/python2.7/site-packages/manticore-0.1.10-py2.7.egg/manticore/manticore.py", line 486, in _hook_callback
    cb(state)
  File "build/bdist.linux-x86_64/egg/deepstate/main_manticore.py", line 255, in <lambda>
    return lambda state: state.invoke_model(func)
  File "/root/.local/lib/python2.7/site-packages/manticore-0.1.10-py2.7.egg/manticore/core/state.py", line 396, in invoke_model
    self._platform.invoke_model(model, prefix_args=(self,))
  File "/root/.local/lib/python2.7/site-packages/manticore-0.1.10-py2.7.egg/manticore/platforms/platform.py", line 35, in invoke_model
    self._function_abi.invoke(model, prefix_args)
  File "/root/.local/lib/python2.7/site-packages/manticore-0.1.10-py2.7.egg/manticore/core/cpu/abstractcpu.py", line 349, in invoke
    result = model(*arguments)
  File "build/bdist.linux-x86_64/egg/deepstate/main_manticore.py", line 155, in hook_IsSymbolicUInt
    return DeepManticore(state).api_is_symbolic_uint(arg)
  File "build/bdist.linux-x86_64/egg/deepstate/common.py", line 417, in api_is_symbolic_uint
    solutions = self.concretize_many(arg, 2)
  File "build/bdist.linux-x86_64/egg/deepstate/main_manticore.py", line 127, in concretize_many
    return self.state.solver.eval_upto(val, max_num)
AttributeError: 'State' object has no attribute 'solver'

ERROR:deepstate.mcore:State 1 terminated due to internal error: 'State' object has no attribute 'solver'
```

error does not terminate run, but results in missing failure case for PrimePolynomial_OnlyGeneratesPrimes_NoStreaming, etc.

Problem also appears (similar symptom) for OneOf example.

**Comment by mossberg** [comment]
https://github.com/trailofbits/deepstate/blob/f61e4b4ff679fc1135019d6c0d7b4cd5e432d4ea/bin/deepstate/main_manticore.py#L126

so it looks like this line is trying to access `.solver` on a State object, but that is actually not part of the State API. (there is also no `eval_upto` on our Solver object). I believe you want something like `return self.state.solve_n(val, max_num)` :)

**Comment by agroce** [comment]
Hmm, making that change switches the failure to:

```
126.38s$ nosetests --verbose
test_basic_functionality (test_basic_functionality.TestBasicFunctionality) ... INFO:deepstate.mcore:Running 2 tests across 1 workers
INFO:deepstate:Running PrimePolynomial_OnlyGeneratesPrimes from /home/travis/build/trailofbits/deepstate/examples/Primes.cpp(30)
CRITICAL:deepstate:/home/travis/build/trailofbits/deepstate/examples/Primes.cpp(38): 40^2 + 40 + 41 is not prime
ERROR:deepstate:Failed: PrimePolynomial_OnlyGeneratesPrimes
INFO:deepstate:Input: 00 00 00 28 00 00 00 29 00 00 00 29
INFO:deepstate:Saving input to out/Primes.cpp/PrimePolynomial_OnlyGeneratesPrimes/79ed996922b59c2f7953d8721c8367be.fail
2018-07-13 22:19:13,378: [7556] m.c.executor:ERROR: Exception: __init__() got an unexpected keyword argument 'procedure'
Traceback (most recent call last):
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/manticore-0.1.10-py2.7.egg/manticore/core/executor.py", line 461, in run
    if not current_state.execute():
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/manticore-0.1.10-py2.7.egg/manticore/core/state.py", line 127, in execute
    result = self._platform.execute()
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/manticore-0.1.10-py2.7.egg/manticore/platforms/linux.py", line 2216, in execute
    self.current.execute()
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/manticore-0.1.10-py2.7.egg/manticore/core/cpu/abstractcpu.py", line 822, in execute
    self._publish('will_execute_instruction', self.PC, insn)
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/manticore-0.1.10-py2.7.egg/manticore/utils/event.py", line 118, in _publish
    self._publish_impl(_name, *args, **kwargs)
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/manticore-0.1.10-py2.7.egg/manticore/utils/event.py", line 134, in _publish_impl
    sink._publish_impl(_name, *args, **kwargs)
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/manticore-0.1.10-py2.7.egg/manticore/utils/event.py", line 134, in _publish_impl
    sink._publish_impl(_name, *args, **kwargs)
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/manticore-0.1.10-py2.7.egg/manticore/utils/event.py", line 132, in _publish_impl
    sink._publish_impl(_name, self, *args, **kwargs)
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/manticore-0.1.10-py2.7.egg/manticore/utils/event.py", line 126, in _publish_impl
    callback(robj(), *args, **kwargs)
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/manticore-0.1.10-py2.7.egg/manticore/manticore.py", line 486, in _hook_callback
    cb(state)
  File "build/bdist.linux-x86_64/egg/deepstate/main_manticore.py", line 254, in <lambda>
    return lambda state: state.invoke_model(func)
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/manticore-0.1.10-py2.7.egg/manticore/core/state.py", line 396, in invoke_model
    self._platform.invoke_model(model, prefix_args=(self,))
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/manticore-0.1.10-py2.7.egg/manticore/platforms/platform.py", line 35, in invoke_model
    self._function_abi.invoke(model, prefix_args)
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/manticore-0.1.10-py2.7.egg/manticore/core/cpu/abstractcpu.py", line 349, in invoke
    result = model(*arguments)
  File "build/bdist.linux-x86_64/egg/deepstate/main_manticore.py", line 232, in hook_MinUInt
    return DeepManticore(procedure=self).api_min_uint(val)
TypeError: __init__() got an unexpected keyword argument 'procedure'
```

**Comment by agroce** [comment]
Yeah, I think it's a confusion between DeepAngr and DeepManticore args

**Comment by agroce** [comment]
Fixed in #67
